### DISCUSSION TITLE
File Reader: S3 bucket connection problems

### DISCUSSION THREAD

**Discussion by JoeFox82** [nextgenhealthcare/connect#5462] [Q&A] [answered]
Hi together,

I've an EC2 Windows instance and want to connect to an S3 bucket with Mirth Connect Version 4.1.1.  Say the Amazon S3 arn is _arn:aws:s3:::mirth-shared_ and the region is _eu-central-1_. I've access to the s3 bucket from the EC2 instance without any credentials. 

When connecting from the local installed aws client, the connections works:
_"C:\Program Files\Amazon\AWSCLIV2\aws.exe" s3 ls s3://mirth-shared_

When connecting from Mirth Connect, the connectiong does not work:
![image](https://user-images.githubusercontent.com/96169329/195054550-17f8980d-1ea7-4413-8f63-bb95f404d794.png)

Does anyone know, what I'm doing wrong?

Kinds regards,
Jochen

**Comment by JoeFox82** [comment]
That's curious. Changing the _Anonymous_ option from _yes_ to _no_ seems to work now:
![image](https://user-images.githubusercontent.com/96169329/195079430-56663cda-aef9-4855-bc1d-0d7cb5fd2729.png)

Have to test it and waiting for files in the bucket...

**Reply by kevinfguo** [reply]
For those looking to connect via ECS, with a properly permissioned task role this was the answer!

**Comment by jonbartels** [comment, marked as answer]
Related discussion in Slack yesterday https://mirthconnect.slack.com/archives/C02SW0K4D/p1665434614433589

**Reply by JoeFox82** [reply]
Sorry, I cannot open the link cause I'm not a slack member :-/

**Reply by jonbartels** [reply]
@JoeFox82 - The links to register with Slack are at https://github.com/nextgenhealthcare/connect#useful-links

**Reply by JoeFox82** [reply]
Thank you very much!
